nf_core/modules/module_test.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+"""
+The ModulesTest class runs the tests locally
+"""
+
+import logging
+import questionary
+import os
+import pytest
+import sys
+import rich
+
+import nf_core.utils
+from .modules_repo import ModulesRepo
+
+log = logging.getLogger(__name__)
+
+class ModulesTest(object):
+    def __init__(
+        self,
+        module_name=None,
+        no_prompts=False,
+        pytest_args="",
+    ):
+        self.module_name = module_name
+        self.no_prompts = no_prompts
+        self.pytest_args = pytest_args
+        self.module_dir = None
+
+    def run(self):
+        """Run test steps"""
+        if not self.no_prompts:
+            log.info(
+                "[yellow]Press enter to use default values [cyan bold](shown in brackets) [yellow]or type your own responses"
+            )
+        self.check_inputs()
+        self.set_profile()
+        self.run_pytests()
+
+    def check_inputs(self):
+        """Do more complex checks about supplied flags."""
+
+        # Get the tool name if not specified
+        if self.module_name is None:
+            modules_repo = ModulesRepo()
+            modules_repo.get_modules_file_tree()
+            self.module_name = questionary.autocomplete(
+                "Tool name:",
+                choices=modules_repo.modules_avail_module_names,
+                style=nf_core.utils.nfcore_question_style,
+            ).ask()
+        self.module_dir = os.path.join("modules", *self.module_name.split("/"))
```

Comment:
I don't really get why the `*self.module_name.split("/")` is necessary. Maybe I lack some experience with modules, but I thought they always live directly in the modules directory and not in sub directories.

And even if, the split with `"/"`  will probably fail on Windows. Better use `from pathlib import Path` and then `self.module_dir =  Path("modules") / self.module_name` (Not 💯 about this thogh, I didn't check what modules_repo.modules_avail_module_names returns. Strongly suspect a string, and if it is the proposed code should work as expected.)

Reply:
The module name can be of the form tool/subtool. For example with aligners like [bwa](https://github.com/nf-core/modules/tree/master/modules/bwa) where you can use "bwa mem", "bwa index", ...

Reply:
Still, pathlib.Path will be more robust code and easier to read. `if not os.path.isdir(self.module_dir):` would then become `if not self.module_dir.is_dir():`

`modules_dir` is only used in this context - is setting the state necessary? If so, it might also be worth documenting it in the docstring.

Should self.module_dir assigned set before it is clear it exists?

Reply:
For me it looks like a check that should be done before setting the module_name attribute. In a way I think setting module_name should fail if module_dir does not exist.

Something along these lines:

```python
>>> from pathlib import Path
>>> class M():
...     def __init__(self, module_name = None):
...         self._module_name = None
...         if module_name:
...             self.module_name = module_name
...     
...     @property
...     def module_name(self):
...         return self._module_name
...     @module_name.setter
...     def module_name(self, module_name):
...         module_dir = Path("modules") / module_name
...         if not module_dir.is_dir():
...             # There could probably a better error message here listing possible module names
...             raise ValueError(f"Module directory not found: {module_dir.absolute()}")
...         self._module_name = module_name
... 
>>> m = M()
>>> m.module_name is None
True
>>> 
>>> 
>>> # Setting module name to a bad value will fail.
... m.module_name = "inexistent"
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 14, in module_name
ValueError: Module directory not found: /path/to/modules/inexistent
>>> 
>>> # It will also fail if it is done at instantiation
... M(module_name = "inexistent")
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 5, in __init__
  File "<stdin>", line 14, in module_name
ValueError: Module directory not found: /path/to/modules/inexistent
>>> 
>>> # Both methods will work if the module exists
... M(module_name = "module").module_name
'module'
>>> m = M()
>>> m.module_name = "module"
>>> m.module_name
'module'
```

I fear I am turning this PR into a marathon. Please feel free to adapt at your own discretion.

Reply:
I applied the suggestion of pathlib. 
Regarding setting module_name after checking module_dir, I am afraid I don't fully get your point. module_name can be set when the user provides the module that needs to be tested, either by command line or by prompt. Afterwards, module_dir is built to check if the module exists. I don't think it's a problem to set module_name before that check. If I am missing something, maybe you could make a code suggestion? I hope it's not asking too much :)

Reply:
You can mark this as resolved if you are happy with the status quo.

Reply:
In that case, it might be better to leave it as it is, so it is not too different from the other tools. Thanks!